frontend/src/metabase/visualizer/components/VisualizerModal/VisualizerModal.tsx

```diff
@@ -33,25 +49,46 @@ export function VisualizerModal({
   const wasOpen = usePrevious(open);
   const dispatch = useDispatch();
 
+  const { modalContent, show: askConfirmation } = useConfirmation();
+
+  const hasChanged = useHasChanged();
```

Comment:
WDYT about extending [visualizer's `getIsDirty` selector](https://github.com/metabase/metabase/blob/18259ef17ffc4cd4dbd7653e82a09c8f5930f763/frontend/src/metabase/visualizer/selectors.ts#L182) to handle this? We'd need to store the initial state to compare to. I think we'd need that later as well to e.g., disable the "Save" button until any changes are made

Reply:
Yeah I can give this a shot